File: src/main/java/org/noxet/noxetserver/playerstate/properties/PSPFoodLevel.java
package org.noxet.noxetserver.playerstate.properties;

import org.bukkit.entity.Player;
import org.noxet.noxetserver.playerstate.PlayerStateProperty;

public class PSPFoodLevel implements PlayerStateProperty<Integer> {
    @Override
    public String getConfigName() {
        return "food_level";
    }

    @Override
    public Integer getDefaultSerializedProperty() {
        return 20;
    }

    @Override
    public Integer getSerializedPropertyFromPlayer(Player player) {
        return player.getFoodLevel();
    }

    @Override
    public void restoreProperty(Player player, Integer level) {
        player.setFoodLevel(level);
    }
}

File: src/main/java/org/noxet/noxetserver/playerstate/properties/PSPFoodLevel.scala
package org.noxet.noxetserver.playerstate.properties

import org.bukkit.entity.Player
import org.noxet.noxetserver.playerstate.PlayerStateProperty

object PSPFoodLevel extends PlayerStateProperty[Integer]:
  override def getConfigName: String = "food_level"

  override def getDefaultSerializedProperty: Integer = 20

  override def getSerializedPropertyFromPlayer(player: Player): Integer =
    player.getFoodLevel

  override def restoreProperty(player: Player, level: Integer): Unit =
    player.setFoodLevel(level)